src/sagemaker/jumpstart/types.py

```diff
@@ -940,10 +1004,340 @@ def to_json(self) -> Dict[str, Any]:
+        "deprecated_message",
+        "deprecate_warn_message",
+        "resource_name_base",
+        "gated_bucket",
+        "training_supported",
+        "incremental_training_supported",
+    ]
+
+    __slots__ = slots + JumpStartMetadataBaseFields.__slots__
+
+    def __init__(  # pylint: disable=super-init-not-called
+        self,
+        component_name: str,
+        component: Optional[Dict[str, Any]],
+    ):
+        """Initializes a JumpStartConfigComponent object from its json representation.
+
+        Args:
+            component_name (str): Name of the component.
+            component (Dict[str, Any]):
+                Dictionary representation of the config component.
+        """
+        self.component_name = component_name
+        self.from_json(component)
+
+    def from_json(self, json_obj: Dict[str, Any]) -> None:
+        """Initializes a JumpStartConfigComponent object from its json representation.
+
+        Args:
+            json_obj (Dict[str, Any]):
+                Dictionary representation of the config component.
+        """
+        for field in json_obj.keys():
+            if field not in self.__slots__:
+                raise ValueError(f"Invalid component field: {field}")
+            setattr(self, field, json_obj[field])
+
+
+class JumpStartMetadataConfig(JumpStartDataHolderType):
+    """Data class of JumpStart metadata config."""
+
+    __slots__ = [
+        "base_fields",
+        "benchmark_metrics",
+        "config_components",
+        "resolved_metadata_config",
+    ]
+
+    def __init__(
+        self,
+        base_fields: Dict[str, Any],
+        config_components: Dict[str, JumpStartConfigComponent],
+        benchmark_metrics: Dict[str, JumpStartBenchmarkStat],
+    ):
+        """Initializes a JumpStartMetadataConfig object from its json representation.
+
+        Args:
+            base_fields (Dict[str, Any]):
+                The default base fields that are used to construct the final resolved config.
+            config_components (Dict[str, JumpStartConfigComponent]):
+                The list of components that are used to construct the resolved config.
+            benchmark_metrics (Dict[str, JumpStartBenchmarkStat]):
+                The dictionary of benchmark metrics with name being the key.
+        """
+        self.base_fields = base_fields
+        self.config_components: Dict[str, JumpStartConfigComponent] = config_components
+        self.benchmark_metrics: Dict[str, JumpStartBenchmarkStat] = benchmark_metrics
+        self.resolved_metadata_config: Optional[Dict[str, Any]] = None
+
+    def to_json(self) -> Dict[str, Any]:
+        """Returns json representation of JumpStartMetadataConfig object."""
+        json_obj = {att: getattr(self, att) for att in self.__slots__ if hasattr(self, att)}
+        return json_obj
+
+    @property
+    def resolved_config(self) -> Dict[str, Any]:
+        """Returns the final config that is resolved from the list of components.
```

Comment:
The components isn't a list, `[...] from the components map.` or `[...] from the dict of components`.